From f36271513241b0df0b78e15c9774bcf63f2d929e Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Thu, 14 Jul 2022 14:10:22 +0200
Subject: [PATCH 1/2] kvm: add support for boolean statistics

The next version of Linux will introduce boolean statistics, which
can only have 0 or 1 values.  Convert them to the new QAPI fields
added in the previous commit.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(backported from commit 105bb7cdbe9b60129034410e228535f5ea5648ea)
[MR: only backport linux-header changes]
Signed-off-by: Matthew Rosato <mjrosato@linux.ibm.com>

Origin: backport, https://gitlab.com/qemu-project/qemu/commit/105bb7cdbe
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1853307
Last-Update: 2023-07-04

---
 linux-headers/linux/kvm.h | 1 +
 1 file changed, 1 insertion(+)

Index: qemu/linux-headers/linux/kvm.h
===================================================================
--- qemu.orig/linux-headers/linux/kvm.h	2023-07-04 20:49:59.000000000 -0400
+++ qemu/linux-headers/linux/kvm.h	2023-07-04 20:51:04.000000000 -0400
@@ -2031,6 +2031,7 @@
 #define KVM_STATS_UNIT_BYTES		(0x1 << KVM_STATS_UNIT_SHIFT)
 #define KVM_STATS_UNIT_SECONDS		(0x2 << KVM_STATS_UNIT_SHIFT)
 #define KVM_STATS_UNIT_CYCLES		(0x3 << KVM_STATS_UNIT_SHIFT)
+#define KVM_STATS_UNIT_BOOLEAN		(0x4 << KVM_STATS_UNIT_SHIFT)
 #define KVM_STATS_UNIT_MAX		KVM_STATS_UNIT_CYCLES
 
 #define KVM_STATS_BASE_SHIFT		8
